### DISCUSSION TITLE
LocalFolder XAML image load issue and its workaround

### DISCUSSION THREAD

**Discussion by Cyroxin** [unoplatform/uno#8905] [General]
Greetings!

Here is an odd issue I've stumbled upon and added a workaround for. 

**XAML ms-appdata**

The following does not work in GTK for windows on the latest 4.3.6 version. However, it does work for UWP:

```XAML
<Page.Content>
    <Image Stretch="Uniform" MinHeight="200" MinWidth="200" Source="ms-appdata:///local/icon.png" />
</Page.Content>
```

____

**Workaround:**

Working around this is possible by using actual code. 

First binding the xaml into the viewmodel of the page class needs to be done like this:
```XAML
<Page.Content>
    <Image Stretch="Uniform" MinHeight="200" MinWidth="200" Source="{x:Bind ViewModel.Icon,Mode=OneWay}" />
</Page.Content>
```
The file needed needs to be acquired. For this purpose it is possible to use these:

```C#
public static StorageFolder LocalFolder = ApplicationData.Current.LocalFolder;
```
```C#
var Files = await LocalFolder.GetFilesAsync();
```
**Note, the above is slightly simplified, in reality it would be taken from a folder within the LocalFolder**

Then finally here is a very dirty snippet for acquiring a Bitmap image that should be bound to within the earlier xaml. I recommend cleaning it up according to your needs since this is all code used for testing purposes only.
```C#
public BitmapImage Icon
		{
			get
			{
				var icon = Files.FirstOrDefault(x => x.Name == "icon.png");
				BitmapImage bitmapImage = new BitmapImage();

				// Load the file from storage
				if (icon != null)
				{
					var queue = DispatcherQueue.GetForCurrentThread();

					var stream = icon.OpenReadAsync();
					stream.AsTask().ContinueWith(r =>
					{
						if ((queue?.TryEnqueue(DispatcherQueuePriority.Normal, () =>
						{
							bitmapImage.SetSourceAsync(r.Result);
							stream.Close();
						}) ?? false) == false)
						{
							try
							{
								bitmapImage.SetSourceAsync(r.Result);
								stream.Close();
							}
							catch { }
						}

					});
				}
				else // Not cached, download. 
				{
                                       //This section can be removed if the file does not have an alternative source in the web.
					var path = Path.Combine(RawBaseUrl, "path","to","file", "icon.png");
					bitmapImage.UriSource = new Uri(path);
				}

				return bitmapImage;
			}
		}
```

**Comment by jeromelaban** [comment]
Thanks for the report. Let's track this with https://github.com/unoplatform/uno/issues/8909, there's also the fact that the Local folder is not pointing to an appropriate location (https://github.com/unoplatform/uno/issues/8910).
